README.md
READ ME
=======

This is PHP implementation of QR Code 2-D barcode generator. It is pure-php
LGPL-licensed implementation based on C libqrencode by Kentaro Fukuchi.

## LICENSING ##

Copyright (C) 2010-2013 by Dominik Dzienia 

This library is free software; you can redistribute it and/or modify it under
the terms of the GNU Lesser General Public License as published by the Free
Software Foundation; either version 3 of the License, or any later version.

This library is distributed in the hope that it will be useful, but WITHOUT ANY
WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR A
PARTICULAR PURPOSE. See the GNU Lesser General Public License (LICENSE file)
for more details.

You should have received a copy of the GNU Lesser General Public License along
with this library; if not, write to the Free Software Foundation, Inc., 51
Franklin St, Fifth Floor, Boston, MA 02110-1301 USA

## INSTALATION AND USAGE ##

 * INSTALL.md file
 * http://sourceforge.net/apps/mediawiki/phpqrcode/index.php?title=Main_Page
 * docs/html/index.html

## CONTACT ##

Fell free to contact me via e-mail (deltalab at poczta dot fm) or using
folowing project pages:

 * http://sourceforge.net/projects/phpqrcode/
 * http://phpqrcode.sourceforge.net/
  
## ACKNOWLEDGMENTS ##

Based on C libqrencode library (ver. 3.1.1) 
Copyright (C) 2006-2010 by Kentaro Fukuchi
http://megaui.net/fukuchi/works/qrencode/index.en.html

QR Code is registered trademarks of DENSO WAVE INCORPORATED in JAPAN and other
countries.

Reed-Solomon code encoder is written by Phil Karn, KA9Q.
Copyright (C) 2002, 2003, 2004, 2006 Phil Karn, KA9Q
 

CHANGELOG.md
CHANGELOG
=========

## 1.9.9 build 20130526

  - __internal__ build (no ZIP released)
  - over 20 full examples, with example browser
  - in-code documentation, doxygen config and HTML Api docs
  - refactored __qrstr__ -> QRspec::set, FrameFiller -> QRframeFiller
  - fixed default parameters in QRCode:: static methods
  - fixed bugs (like string instead of $string, wrong parameters) found by community
  - improved debug output from QRarea and QRspec::debug

## 1.1.4 build 2010100721

  - added missing static keyword QRinput::check (found by Luke Brookhart, Onjax LLC)
  
## 1.1.3 build 2010081807

  - short opening tags replaced with standard ones
  
## 1.1.2 build 2010032722

  - full integration with TCPDF thanks to Nicola Asuni, it's author
  - fixed bug with alphanumeric encoding detection

## 1.1.1 build 2010032405

  - patch by Rick Seymour allowing saving PNG and displaying it at the same time
  - added version info in VERSION file
  - modified merge tool to include version info into generated file
  - fixed e-mail in almost all head comments

## 1.1.0 build 2010032113

  - added merge tool wich generate merged version of code
    located in phpqrcode.php
  - splited qrconst.php from qrlib.php
  
## 1.0.0 build 2010031920 

  - first public release
  - help in readme, install
  - cleanup ans separation of QRtools and QRspec
  - now TCPDF binding requires minimal changes in TCPDF, having most of job
    done in QRtools tcpdfBarcodeArray
  - nicer QRtools::timeBenchmark output
  - license and copyright notices in files
  - indent cleanup - from tab to 4spc, keep it that way please :)
  - sf project, repository, wiki
  - simple code generator in index.php


index.php
<?php    
/*
 * PHP QR Code encoder
 *
 * Exemplatory usage
 *
 * PHP QR Code is distributed under LGPL 3
 * Copyright (C) 2010-2013 Dominik Dzienia <deltalab at poczta dot fm>
 *
 * This library is free software; you can redistribute it and/or
 * modify it under the terms of the GNU Lesser General Public
 * License as published by the Free Software Foundation; either
 * version 3 of the License, or any later version.
 *
 * This library is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
 * Lesser General Public License for more details.
 *
 * You should have received a copy of the GNU Lesser General Public
 * License along with this library; if not, write to the Free Software
 * Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA
 */
 
    echo '<!DOCTYPE html>
    <html>
    <head>
        <meta charset="utf-8">
        <title>PHP QR Code Demo</title>
        <script type="text/javascript" src="lib/js/qrcanvas.js"></script>
    </head>
    <body>
    <h1>PHP QR Code</h1>
    <hr/>';
    
    echo '<a href="examples/index.php">Examples and Demos</a> (<a href="">online</a>) 
    | <a href="docs/html/index.html">API Documentation</a> (<a href="">online</a>)  
    | <a href="http://sourceforge.net/projects/phpqrcode/">Online SF project</a> 
    | <a href="http://phpqrcode.sourceforge.net/">Online SF Main Site</a> 
    | <a href="http://sourceforge.net/apps/mediawiki/phpqrcode/">Online SF wiki</a>
    | <a href="http://sourceforge.net/donate/index.php?group_id=311533">Donate!</a><hr />';
    
    // setup and input processing ----------------------------------------------
    
    //set it to writable location, a place for temp generated PNG files
    $FILE_TEMP_DIR = dirname(__FILE__).DIRECTORY_SEPARATOR.'temp'.DIRECTORY_SEPARATOR;
    
    //html PNG location prefix
    $FILE_WEB_DIR = 'temp/';

    include "lib/full/qrlib.php";    
    
    //ofcourse we need rights to create temp dir
    if (!file_exists($FILE_TEMP_DIR))
        mkdir($FILE_TEMP_DIR);
    

    //processing form input
    //remember to sanitize user input in real-life solution !!!
    
    $errorCorrectionLevel = 'L';
    if (isset($_REQUEST['level']) && in_array($_REQUEST['level'], array('L','M','Q','H')))
        $errorCorrectionLevel = $_REQUEST['level'];    

    $matrixPointSize = 4;
    if (isset($_REQUEST['size']))
        $matrixPointSize = min(max((int)$_REQUEST['size'], 1), 10);

    $textData = 'PHP QR Code :)';
    
    if (isset($_REQUEST['data']) && (trim($_REQUEST['data']) != '')) { 
        $textData = $_REQUEST['data'];
    }
        
    //config form --------------------------------------------------------------
    
    echo '<form action="index.php" method="post">
        Data:&#160;<input name="data" value="'.(isset($_REQUEST['data'])?htmlspecialchars($_REQUEST['data']):'PHP QR Code :)').'" />&#160;
        ECC:&#160;<select name="level">
            <option value="L"'.(($errorCorrectionLevel=='L')?' selected="true"':'').'>L - smallest</option>
            <option value="M"'.(($errorCorrectionLevel=='M')?' selected="true"':'').'>M</option>
            <option value="Q"'.(($errorCorrectionLevel=='Q')?' selected="true"':'').'>Q</option>
            <option value="H"'.(($errorCorrectionLevel=='H')?' selected="true"':'').'>H - best</option>
        </select>&#160;
        Size:&#160;<select name="size">';
        
    for($i=1;$i<=10;$i++)
        echo '<option value="'.$i.'"'.(($matrixPointSize==$i)?' selected="true"':'').'>'.$i.'</option>';
        
    echo '</select>&#160;
        <input type="submit" value="GENERATE" /></form><hr/>';

    //display generated file ---------------------------------------------------
    
    $pngFilename = $FILE_TEMP_DIR.'test'.md5($textData.'|'.$errorCorrectionLevel.'|'.$matrixPointSize).'.png';
    QRcode::png($textData, $pngFilename, $errorCorrectionLevel, $matrixPointSize, 2);    
    
    echo '<img src="'.$FILE_WEB_DIR.basename($pngFilename).'" /><hr />';

    // benchmark ---------------------------------------------------------------
    
    QRtools::timeBenchmark();    
    
    // links ---------------------------------------------------------------
    
    echo '<hr />';

    echo '</body></html>';